xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/PropertyClass.java

```diff
@@ -191,10 +191,28 @@ public String toString(BaseProperty property)
 
     @Override
     public BaseProperty fromString(String value)
+    {
+        return fromString(value, null);
+    }
+
+    @Override
+    public BaseProperty fromString(String value, BaseProperty baseProperty)
     {
         return null;
     }
 
+    /**
+     * Retrieve the already existing base property based on current name, or create a new one.
+     *
+     * @return the existing {@link BaseProperty} or a new one.
+     * @since 17.3.0RC1
+     */
+    @Unstable
+    protected BaseProperty getCurrentOrNewProperty(BaseProperty property)
+    {
+        return (property != null) ? property : newProperty();
```

Comment:
It might be outside the scope of this issue, but I'm wondering if the xclass should "auto-repair" the property if it does not have the right type. Right now, the following code might fail if `getCurrentOrNewProperty` gives a property of the wrong type (compared to the previous situation where that code was expecting to always get a new property).